### PR TITLE
net-irc/emech: fix build for clang16

### PR DESCRIPTION
**Pull request by Schievel1**
Still has error 
```
debug.c:199:3: error: initializer element is not a compile-time constant
{ (int)tcl_timer_jump,  "tcl_timer_jump"        },
```

when built with USE="debug"

Signed-off-by: Pascal Jäger <pascal.jaeger@leimstift.de>

### CODE REVIEW

File: net-irc/emech/emech-3.0.99_p3-r2.ebuild [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+		-e '/^GDBFLAG/d' \
+		-e '/^PIPEFLAG/d' \
+		src/Makefile.in || die
+
+	eapply_user
+}
+
+myconf() {
+	tc-export CC
+	echo ./configure $*
```
Comment (thesamesam) [inline net-irc/emech/emech-3.0.99_p3-r2.ebuild:43]: Replace these two lines with edo, and fold myconf() into src_configure (just use an array)

File: net-irc/emech/emech-3.0.99_p3-r2.ebuild [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+IUSE="debug session tcl"
+KEYWORDS="~amd64 ~ppc ~x86"
+
+RDEPEND="virtual/libcrypt:="
+DEPEND="${RDEPEND}"
+
+DOC_CONTENTS="You can find a compressed sample config file at /usr/share/doc/${PF}"
+
+src_prepare() {
+	eapply "${FILESDIR}/${P}-buildfix.patch"
```
Comment (thesamesam) [inline net-irc/emech/emech-3.0.99_p3-r2.ebuild:24]: Use PATCHES=() instead

File: net-irc/emech/emech-3.0.99_p3-r2.ebuild [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+		-e 's: "messages/":"/usr/share/energymech/messages/":' \
+		src/config.h.in || die
+	# Respect CFLAGS and LDFLAGS
+	sed -i \
+		-e '/^LFLAGS/s/\$(PIPEFLAG)/\0 \$(OPTIMIZE) \$(LDFLAGS)/' \
+		-e '/^GDBFLAG/d' \
+		-e '/^PIPEFLAG/d' \
+		src/Makefile.in || die
+
+	eapply_user
```
Comment (thesamesam) [inline net-irc/emech/emech-3.0.99_p3-r2.ebuild:38]: s/eapply_user/default, and put it before all the seds

**Comment by Schievel1** [comment]
Works fine with a version bump. Just a little laugh:
```
int is_safepath(const char *path)
{
	struct stat st;
	ino_t	ino;
	char	tmp[PATH_MAX];
	const char *src;
	char	*dst;

	int path_token_check(void)
	{
		*dst = 0;
		lstat(tmp,&st);
		if (!(S_ISREG(st.st_mode) || S_ISDIR(st.st_mode)))
			return(FALSE);
		if (st.st_ino == parent_inode)
			return(FALSE);
		return(S_ISDIR(st.st_mode) ? 2 : 1);
	}

	if (*(src = path) == '/')
		return(FALSE);

	dst = tmp;
	while(*src)
	{
		if (*src == '/' && !path_token_check())
			return(FALSE);
		if (dst == tmp + PATH_MAX-1)
			return(FALSE);
		*dst++ = *src++;
	}
	return(path_token_check() == 1 ? TRUE : FALSE);
}
```

It puzzles me that GCC just went over this without even issues a warning... :D

**Review by juippis — APPROVED** [review]
Seems to work. One weird note though:
```
checking for perl ... ./configure: line 947: perl: command not found
```

**Comment by juippis** [comment]
Well, also, 
```
 * SED: the following did not cause any changes
 *     sed -i -e 's: "help/":"/usr/share/energymech/help/":' -e 's: "messages/":"/usr/share/energymech/messages/":' src/config.h.in || die;
 * no-op: -e s: "help/":"/usr/share/energymech/help/":
 * no-op: -e s: "messages/":"/usr/share/energymech/messages/":
```
since you seem to be doing more than just adding a patch, maybe take a look at these?

**Comment by Schievel1** [comment]
> Well, also,
> 
> ```
>  * SED: the following did not cause any changes
>  *     sed -i -e 's: "help/":"/usr/share/energymech/help/":' -e 's: "messages/":"/usr/share/energymech/messages/":' src/config.h.in || die;
>  * no-op: -e s: "help/":"/usr/share/energymech/help/":
>  * no-op: -e s: "messages/":"/usr/share/energymech/messages/":
> ```
> 
> since you seem to be doing more than just adding a patch, maybe take a look at these?

Will do, but most importantly: What do you use to see that sed does nothing? I need this. :D

**Comment by juippis** [comment]
> Will do, but most importantly: What do you use to see that sed does nothing? I need this. :D

https://github.com/ionenwks/iwdevtools#qa-sed

**Comment by Schievel1** [comment]
> > Will do, but most importantly: What do you use to see that sed does nothing? I need this. :D
> 
> https://github.com/ionenwks/iwdevtools#qa-sed

Thanks, thats really useful.

File: net-irc/emech/emech-3.0.99_p20221123.ebuild [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+		--without-uptime
+		--with-web
+		--with-wingate
+		--without-profiling
+		--without-redirect
+		$(use_with tcl)
+		$(use_with session)
+		$(use_with debug)
+	)
+	./configure "${myconf[@]}" || die "Configure failed"
```
Comment (juippis) [inline net-irc/emech/emech-3.0.99_p20221123.ebuild:71]: One more thing: why `./configure` and not `econf`? Also there's still this: 
```
checking for perl ... ./configure: line 947: perl: command not found
no
```
could you check what it tries to do?

Comment (Schievel1) [reply net-irc/emech/emech-3.0.99_p20221123.ebuild:71]: Well for me it just says `checking for perl ... yes` there. 
It does this: `pinc1=`perl -MConfig -e 'print $Config{archlib}'``

If it still works for you, I would not add this to dependencies.

**Review by juippis — APPROVED** [review]
Thanks!

**Pull request closed without merging**